wikitextprocessor/core.py

```diff
@@ -973,9 +973,7 @@ def analyze_templates(self) -> None:
             "Analyzing which templates should be expanded before parsing"
         )
         # Add/overwrite templates
-        template_ns: NamespaceDataEntry = self.NAMESPACE_DATA.get(
-            "Template", EMPTY_NAMESPACEDATA
```

Comment:
Please stop removing annotations when they are put there as documentation.

Reply:
This type annotation is unnecessary because you already added the type to `NAMESPACE_DATA`.

Reply:
It is there as a comment. Just leave them be. We have already talked about this, annotations are not just for the type checker, it also helps to comment the code.

Reply:
OK, I restored it. I just feel this is kind obvious unlike some complicated cases and type checker get not problem to find the type so I deleted it.

Reply:
Thanks. Named types are named for a reason, either they're complicated, like NamespaceDataEntry which is a pretty big TypedDict, or they're simple but it's helpful to know what paths stuff is supposed to take. A `str` is not necessarily just a `str`, and type annotations and variable names are helpful to telling us the *whys* and the *wheres*.

In general, if it's not a basic Python function that produces sometimes immediately obvious like a str or int, the named type annontations are helpful, and they never cause harm. They're also useful as sentinels to keep an eye out on expected values and actual values.

Reply:
I usually won't bothered to add type to value returned from a typed function... Tools in text editors are pretty good for finding errors and code completion.